include/seastar/json/formatter.hh

```diff
@@ -85,31 +115,58 @@ class formatter {
         return to_json(t);
     }
 
-    template<typename K, typename V>
-    static future<> write(output_stream<char>& stream, state s, const std::pair<K, V>& p) {
+    template<typename T>
+    requires (internal::is_pair_like<std::decay_t<T>> &&
+              std::is_rvalue_reference_v<T>)
+    static future<> write(output_stream<char>& stream, state s, T p) {
```

Comment:
Why can't this look like

```c++
template<typename T>
requires (internal::is_pair_like T>
static future<> write(output_stream<char>& stream, state s, T&& p) {
    ...
```
?

Reply:
no reasons. i was trying to consolidate these two variants of `write(...)` overloads for `is_pair_like` types, but ended up giving up on this idea, because it hurts the readability, and went with partially repeating them.

and yeah,  `T&& p` is way better. thank you! i switched to the proposed function signature in the latest revision.